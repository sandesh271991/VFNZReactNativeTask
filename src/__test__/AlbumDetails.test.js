import React, { Component } from 'react';
import 'react-native';
import { shallow } from 'enzyme';
import __AlbumDetailsView from '../component/AlbumDetailsComponent/albumDetailsView';

describe('Album Details Screen', () => {

  it('should render Album Details component', () => {
    const wrapper = shallow(
        <__AlbumDetailsView navigation={{ getParam: jest.fn() }} />
    );
  });

  it('should render initial layout', () => {
  // when
  const component = shallow(<__AlbumDetailsView navigation={{ getParam: jest.fn() }} />);
  // then
  expect(component.getElements()).toMatchSnapshot();
  });

  it('should check if BackButton exists', () => {
    const wrapper = shallow(<__AlbumDetailsView navigation={{ getParam: jest.fn() }} />);
    wrapper.findWhere(n => n.name() === 'Button' && n.prop('testID') === 'HomeButton')
  });
  
});