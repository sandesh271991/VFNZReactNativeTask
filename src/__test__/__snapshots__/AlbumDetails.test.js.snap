// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Album Details Screen should render initial layout 1`] = `
Array [
  <React.Fragment>
    <Provider>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "height": 80,
            "justifyContent": "center",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#dedede",
              "borderRadius": 30,
              "flex": 1,
              "justifyContent": "center",
              "margin": 30,
              "marginTop": 20,
              "width": "80%",
            }
          }
        >
          <Image
            source={
              Object {
                "uri": undefined,
              }
            }
            style={
              Object {
                "backgroundColor": "black",
                "height": 250,
                "width": 250,
              }
            }
          />
          <withTheme(Text)
            style={
              Object {
                "backgroundColor": "#0c82f3",
                "borderRadius": 7,
                "color": "white",
                "fontSize": 23,
                "margin": 7,
                "overflow": "hidden",
                "padding": 10,
                "textAlign": "left",
              }
            }
          />
          <withTheme(Button)
            mode="contained"
            onPress={[Function]}
            style={
              Object {
                "backgroundColor": "blue",
                "height": 40,
                "marginTop": 50,
                "width": "93%",
              }
            }
            testID="HomeButton"
          >
            HOME
          </withTheme(Button)>
        </View>
      </View>
    </Provider>
  </React.Fragment>,
]
`;
